src/libmongoc/tests/test-mongoc-topology.c

```diff
@@ -1972,10 +1972,14 @@ test_request_scan_on_error ()
                 NULL /* server_err */);
    /* with a "not master" error code but a "node is recovery" message, it is
     * considered a "node is recovering" error */
-   TEST_BOTH ("{'ok': 0, 'code': 10107, 'errmsg': 'node is recovering'}",
+   TEST_SINGLE ("{'ok': 0, 'code': 10107, 'errmsg': 'node is recovering'}",
```

Comment:
This relates to [this section of SDAM](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#not-master-and-node-is-recovering):

> For single-threaded clients, in the case of a "not master" or "node is shutting down" error, the client MUST mark the topology as "stale" so the next server selection scans all servers. For a "node is recovering" error, single-threaded clients MUST NOT mark the topology as "stale". If a node is recovering for some time, an immediate scan may not gain useful information.